prometheus_test.go

```diff
@@ -290,90 +292,156 @@ func TestHistogramUnorderedBucketBounds(t *testing.T) {
 	}
 }
 
-func TestConstLabelsIncluded(t *testing.T) {
-	constLabels := prometheus.Labels{
-		"service": "spanner",
-	}
+func TestConstLabelsAndResource(t *testing.T) {
+	testCases := []struct {
+		name        string
+		constLabels prometheus.Labels
+		resource    *resource.Resource
+		want        string
+	}{{
+		name: "neither const labels nor resource",
+		want: `# HELP tests_bar bar
+# TYPE tests_bar counter
+tests_bar{method="issue961"} 1
+# HELP tests_baz baz
+# TYPE tests_baz counter
+tests_baz{method="issue961"} 1
+# HELP tests_foo foo
+# TYPE tests_foo counter
+tests_foo{method="issue961"} 1
+`,
+	}, {
+		name:        "const labels only",
+		constLabels: prometheus.Labels{"service": "spanner"},
+		want: `# HELP tests_bar bar
+# TYPE tests_bar counter
+tests_bar{method="issue961",service="spanner"} 1
+# HELP tests_baz baz
+# TYPE tests_baz counter
+tests_baz{method="issue961",service="spanner"} 1
+# HELP tests_foo foo
+# TYPE tests_foo counter
+tests_foo{method="issue961",service="spanner"} 1
+`,
+	}, {
+		name:     "resource only",
+		resource: &resource.Resource{Type: "test resource", Labels: map[string]string{"region": "us-east"}},
+		want: `# HELP tests_bar bar
+# TYPE tests_bar counter
+tests_bar{method="issue961",region="us-east"} 1
+# HELP tests_baz baz
+# TYPE tests_baz counter
+tests_baz{method="issue961",region="us-east"} 1
+# HELP tests_foo foo
+# TYPE tests_foo counter
+tests_foo{method="issue961",region="us-east"} 1
+`,
+	}, {
+		name:        "both const labels and resource",
+		constLabels: prometheus.Labels{"service": "spanner"},
+		resource:    &resource.Resource{Type: "test resource", Labels: map[string]string{"region": "us-east"}},
+		want: `# HELP tests_bar bar
+# TYPE tests_bar counter
+tests_bar{method="issue961",region="us-east",service="spanner"} 1
+# HELP tests_baz baz
+# TYPE tests_baz counter
+tests_baz{method="issue961",region="us-east",service="spanner"} 1
+# HELP tests_foo foo
+# TYPE tests_foo counter
+tests_foo{method="issue961",region="us-east",service="spanner"} 1
+`,
+	}, {
+		name:        "const labels and resource overlap",
+		constLabels: prometheus.Labels{"service": "spanner"},
+		resource:    &resource.Resource{Type: "test resource", Labels: map[string]string{"service": "bigtable"}},
```

Comment:
I would update this test with non-overlapping labels in resource and const labels.

Reply:
The test case above this one is for non-overlapping labels.

Reply:
It is good to have a test with mix of overlapping and non-overlapping. Hence the suggestion.

Reply:
Done.  Add a test case for that.